src/internal_events/template.rs

```diff
@@ -0,0 +1,31 @@
+use super::InternalEvent;
+use crate::template::TemplateRenderingError;
+use metrics::counter;
+
+pub struct TemplateRenderingFailed<'a> {
+    pub field: Option<&'a str>,
+    pub drop_event: bool,
+    pub error: TemplateRenderingError,
+}
+
+impl<'a> InternalEvent for TemplateRenderingFailed<'a> {
+    fn emit_logs(&self) {
+        let mut msg = "Failed to render template".to_owned();
+        if let Some(field) = self.field {
+            msg.push_str(&format!(" for \"{}\"", field));
```

Comment:
You can use the `Write` interface for the string here instead of creating a temporary:
```suggestion
            use std::fmt::Write;
            let _ = write!(msg, " for \"{}\"", field);
```

Reply:
Great! Thanks!